ipld/read_test.go

```diff
@@ -290,47 +241,50 @@ func TestGetLeavesByNamespace_AbsentNamespaceId(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			dag, dah := putErasuredDataToDag(t, tt.data)
-
-			assertNoRowContainsNID(t, dag, dah, tt.missingNid)
+			eds, err := PutData(ctx, shares, dag)
+			require.NoError(t, err)
+			assertNoRowContainsNID(t, dag, eds, tt.missingNid)
 		})
 	}
 }
 
 func TestGetLeavesByNamespace_MultipleRowsContainingSameNamespaceId(t *testing.T) {
-	t.Run("Same namespace id across multiple rows", func(t *testing.T) {
-		rawData := RandNamespacedShares(t, 16).Raw()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
use `t.Cleanup(cancel)` pls

Reply:
We use `t.Cleanup` in tests instead of defer usually, so that's why i mentioned it. Our code is kind of all over the place with using defer / t.Cleanup so whatever.

Reply:
Yeah, my code uses defer, yours uses t.Cleanup, so yeah ,we need to find a consensus on what we are using to be consistent.
> Our code is kind of all over the place with using defer / t.Cleanup so whatever

Reply:
I started using `t.Cleanup` in tests when you suggested it 😂  but I'm fine with either. Same thing, but we need to start being consistent.

Reply:
True, but not for context, iirc
> I started using t.Cleanup in tests when you suggested it

Reply:
Issued an issue https://github.com/celestiaorg/celestia-node/issues/668